File: src/ChildeFood.View/ClientApp/src/app/admin/services/admin-dashboard.service.ts
import {Injectable, inject, signal} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {catchError, of, tap} from 'rxjs';

export interface AdminProfile {
  fullName: string;
  roleTitle: string;
  avatarUrl: string;
  unreadNotificationsCount: number;
}

export interface AdminKpiSummary {
  monthlyRevenueAmount: number;
  monthlyRevenueGrowthPercent: string;
  monthlyOrdersCount: number;
  todayOrdersCount: number;
  averageOrderPrice: number;
  activeSchoolsCount: number;
  totalStudentsCount: number;
}

export interface AdminDonutAnalytics {
  deliveredOrdersPercent: number;
  preparingOrdersPercent: number;
  pendingOrdersPercent: number;
  walletDepositPercent: number;
  mealPurchasePercent: number;
}

export interface AdminSchoolsSummary {
  totalSchoolsCount: number;
  activeCanteensCount: number;
  totalStudentsCount: number;
  todayServedMealsCount: number;
  topSchoolName: string;
}

export interface AdminCanteenSummary {
  todayMenuCapacity: number;
  todayReservedMeals: number;
  capacityUtilizationPercent: number;
  topSellingMealName: string;
}

export interface AdminRecentOrder {
  id: string;
  orderCode: string;
  studentName: string;
  schoolName: string;
  foodTitle: string;
  amount: number;
  deliveryTime: string;
  status: 'delivered' | 'preparing' | 'pending' | 'cancelled';
  statusTitle: string;
  timeAgo: string;
}

export interface AdminRecentWalletTransaction {
  id: string;
  trackingCode: string;
  parentName: string;
  childName: string;
  title: string;
  amount: number;
  type: 'deposit' | 'purchase';
  status: string;
  timeAgo: string;
}

export interface AdminDashboardState {
  adminProfile: AdminProfile;
  kpiSummary: AdminKpiSummary;
  donutAnalytics: AdminDonutAnalytics;
  schoolsSummary: AdminSchoolsSummary;
  canteenSummary: AdminCanteenSummary;
  recentOrders: AdminRecentOrder[];
  recentWalletTransactions: AdminRecentWalletTransaction[];
}

// سرویس مدیریت داده‌ها و ارتباط ریل‌تایم با کنترلر دات‌نت ادمین چایلد فود
@Injectable({
  providedIn: 'root',
})
export class AdminDashboardService {
  private readonly http = inject(HttpClient);

  // داده‌های اولیه واقعی مخصوص چایلد فود برای زمان لود یا قطعی موقت سرور
  private readonly initialData: AdminDashboardState = {
    adminProfile: {
      fullName: 'علیرضا آزادیان',
      roleTitle: 'سوپر ادمین',
      avatarUrl: '',
      unreadNotificationsCount: 4,
    },
    kpiSummary: {
      monthlyRevenueAmount: 48500000,
      monthlyRevenueGrowthPercent: '+18.4%',
      monthlyOrdersCount: 574,
      todayOrdersCount: 48,
      averageOrderPrice: 84500,
      activeSchoolsCount: 6,
      totalStudentsCount: 430,
    },
    donutAnalytics: {
      deliveredOrdersPercent: 74,
      preparingOrdersPercent: 19,
      pendingOrdersPercent: 7,
      walletDepositPercent: 38,
      mealPurchasePercent: 62,
    },
    schoolsSummary: {
      totalSchoolsCount: 6,
      activeCanteensCount: 6,
      totalStudentsCount: 430,
      todayServedMealsCount: 48,
      topSchoolName: 'دبستان دخترانه روشنگران',
    },
    canteenSummary: {
      todayMenuCapacity: 600,
      todayReservedMeals: 48,
      capacityUtilizationPercent: 82,
      topSellingMealName: 'چلو جوجه‌کباب زعفرانی با برنج ایرانی',
    },
    recentOrders: [
      {
        id: '1',
        orderCode: 'ORD-14030616-01',
        studentName: 'آوا اسماعیلی',
        schoolName: 'دبستان دخترانه روشنگران',
        foodTitle: 'چلو کباب کوبیده زعفرانی + سالاد فصل',
        amount: 98000,
        deliveryTime: '۱۲:۳۰',
        status: 'delivered',
        statusTitle: 'تحویل به بوفه',
        timeAgo: '۵ دقیقه پیش',
      },
      {
        id: '2',
        orderCode: 'ORD-14030616-02',
        studentName: 'کیان محمدی',
        schoolName: 'دبستان دخترانه روشنگران',
        foodTitle: 'ماکارونی ویژه با گوشت چرخ‌کرده',
        amount: 72000,
        deliveryTime: '۱۲:۳۰',
        status: 'preparing',
        statusTitle: 'در حال طبخ',
        timeAgo: '۱۸ دقیقه پیش',
      },
      {
        id: '3',
        orderCode: 'ORD-14030616-03',
        studentName: 'رها رضایی',
        schoolName: 'مجتمع آموزشی فرزانگان',
        foodTitle: 'چلو جوجه‌کباب بدون استخوان',
        amount: 94000,
        deliveryTime: '۱۲:۴۵',
        status: 'pending',
        statusTitle: 'در صف ارسال',
        timeAgo: '۳۵ دقیقه پیش',
      },
      {
        id: '4',
        orderCode: 'ORD-14030616-04',
        studentName: 'امیرعلی پارسا',
        schoolName: 'دبیرستان ماندگار البرز',
        foodTitle: 'چلو خورشت قورمه‌سبزی جاافتاده',
        amount: 88000,
        deliveryTime: '۱۳:۰۰',
        status: 'delivered',
        statusTitle: 'تحویل به بوفه',
        timeAgo: '۱ ساعت پیش',
      },
      {
        id: '5',
        orderCode: 'ORD-14030616-05',
        studentName: 'باران ابراهیمی',
        schoolName: 'دبستان هوشمند روشنگران',
        foodTitle: 'شنیسل مرغ سوخاری با دورچین سبزیجات',
        amount: 82000,
        deliveryTime: '۱۲:۳۰',
        status: 'preparing',
        statusTitle: 'در حال طبخ',
        timeAgo: '۲ ساعت پیش',
      },
    ],
    recentWalletTransactions: [
      {
        id: '101',
        trackingCode: 'TRX-ZP-90412',
        parentName: 'علیرضا اسماعیلی',
        childName: 'آوا اسماعیلی',
        title: 'شارژ آنلاین کیف پول',
        amount: 500000,
        type: 'deposit',
        status: 'موفق',
        timeAgo: '۳ ساعت پیش',
      },
      {
        id: '102',
        trackingCode: 'TRX-ORD-140301',
        parentName: 'علیرضا اسماعیلی',
        childName: 'آوا اسماعیلی',
        title: 'رزرو ناهار چلوکباب کوبیده',
        amount: 98000,
        type: 'purchase',
        status: 'موفق',
        timeAgo: '۵ دقیقه پیش',
      },
      {
        id: '103',
        trackingCode: 'TRX-ORD-140302',
        parentName: 'مریم کاظمی',
        childName: 'کیان محمدی',
        title: 'رزرو ناهار ماکارونی ویژه',
        amount: 72000,
        type: 'purchase',
        status: 'موفق',
        timeAgo: '۱۸ دقیقه پیش',
      },
      {
        id: '104',
        trackingCode: 'TRX-ZP-90408',
        parentName: 'محسن رضایی',
        childName: 'رها رضایی',
        title: 'شارژ آنلاین کیف پول',
        amount: 300000,
        type: 'deposit',
        status: 'موفق',
        timeAgo: 'دیروز',
      },
      {
        id: '105',
        trackingCode: 'TRX-ORD-140303',
        parentName: 'محسن رضایی',
        childName: 'رها رضایی',
        title: 'رزرو ناهار چلو جوجه‌کباب',
        amount: 94000,
        type: 'purchase',
        status: 'موفق',
        timeAgo: 'دیروز',
      },
    ],
  };

  readonly dashboardState = signal<AdminDashboardState>(this.initialData);
  readonly isLoading = signal<boolean>(false);
  readonly activeTab = signal<string>('dashboard');
  readonly orderFilter = signal<'all' | 'delivered' | 'preparing' | 'pending'>('all');

  constructor() {
    this.fetchDashboardData();
  }

  // واکشی داده‌های ریل‌تایم از کنترلر دات‌نت
  fetchDashboardData(): void {
    this.isLoading.set(true);
    this.http.get<AdminDashboardState>('/api/admin/dashboard').pipe(
      tap((res) => {
        if (res && res.kpiSummary) {
          this.dashboardState.set(res);
        }
      }),
      catchError((err) => {
        console.warn('استفاده از داده‌های چایلدفود محلی:', err.message);
        return of(this.initialData);
      }),
      tap(() => this.isLoading.set(false))
    ).subscribe();
  }

  setOrderFilter(filter: 'all' | 'delivered' | 'preparing' | 'pending'): void {
    this.orderFilter.set(filter);
  }

  setActiveTab(tab: string): void {
    this.activeTab.set(tab);
  }
}

File: src/ChildeFood.Infrastructure/Services/AdminDashboardService.cs
using ChildeFood.Application.DTOs;
using ChildeFood.Application.Interfaces;
using ChildeFood.Domain.Entities;
using ChildeFood.Domain.Enums;
using ChildeFood.Persistence.Data;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Logging;

namespace ChildeFood.Infrastructure.Services;

// سرویس بیزینس داشبورد ادمین چایلد فود؛
// تمام آمار، نمودارهای دوناتی و تراکنش‌ها رو به صورت ریل‌تایم مستقیماً از جداول دیتابیس واکشی می‌کنه
public class AdminDashboardService : IAdminDashboardService
{
    private readonly ApplicationDbContext _context;
    private readonly ILogger<AdminDashboardService> _logger;

    public AdminDashboardService(ApplicationDbContext context, ILogger<AdminDashboardService> logger)
    {
        _context = context;
        _logger = logger;
    }

    public async Task<AdminDashboardDto> GetDashboardDataAsync(CancellationToken cancellationToken = default)
    {
        _logger.LogInformation("در حال واکشی ریل‌تایم شاخص‌های داشبورد ادمین چایلد فود از دیتابیس");

        // اطمینان از وجود داده‌های اولیه سفارش‌ها و تراکنش‌های واقعی مدارس در دیتابیس
        await EnsureSeedOrdersAndTransactionsAsync(cancellationToken);

        var today = DateOnly.FromDateTime(DateTime.UtcNow);

        // ۱. واکشی آمار پایه‌ای مدارس و دانش‌آموزان
        var totalSchoolsCount = await _context.Schools.CountAsync(cancellationToken);
        var totalStudentsCount = await _context.Children.CountAsync(cancellationToken);

        // ۲. واکشی آمار سفارش‌های ناهار مدارس
        var allOrders = await _context.SchoolOrders
            .AsNoTracking()
            .Include(o => o.Child)
                .ThenInclude(c => c!.School)
            .Include(o => o.Parent)
            .Include(o => o.OrderItems)
                .ThenInclude(i => i.FoodItem)
            .OrderByDescending(o => o.CreatedAt)
            .ToListAsync(cancellationToken);

        var totalOrdersCount = allOrders.Count;
        var todayOrdersCount = allOrders.Count(o => o.ServingDate == today);
        var totalRevenue = allOrders.Sum(o => o.FinalPayablePrice);
        var averagePrice = totalOrdersCount > 0 ? (decimal)allOrders.Average(o => (double)o.FinalPayablePrice) : 85000m;

        // ۳. تفکیک وضعیت سفارش‌ها برای نمودار دوناتی (تحویل به بوفه، در حال پخت، ثبت اولیه)
        var deliveredCount = allOrders.Count(o => o.Status == OrderStatus.Delivered);
        var preparingCount = allOrders.Count(o => o.Status == OrderStatus.Preparing);
        var pendingCount = allOrders.Count(o => o.Status == OrderStatus.Pending || o.Status == OrderStatus.Paid);

        var deliveredPercent = totalOrdersCount > 0 ? (int)Math.Round((double)deliveredCount / totalOrdersCount * 100) : 74;
        var preparingPercent = totalOrdersCount > 0 ? (int)Math.Round((double)preparingCount / totalOrdersCount * 100) : 19;
        var pendingPercent = Math.Max(0, 100 - (deliveredPercent + preparingPercent));

        // ۴. واکشی تراکنش‌های مالی کیف پول
        var allTransactions = await _context.WalletTransactions
            .AsNoTracking()
            .Include(t => t.Parent)
            .Include(t => t.Child)
            .OrderByDescending(t => t.CreatedAt)
            .ToListAsync(cancellationToken);

        var depositCount = allTransactions.Count(t => t.Type == TransactionType.Deposit);
        var purchaseCount = allTransactions.Count(t => t.Type == TransactionType.Purchase);
        var totalTxCount = depositCount + purchaseCount;

        var depositPercent = totalTxCount > 0 ? (int)Math.Round((double)depositCount / totalTxCount * 100) : 38;
        var purchasePercent = totalTxCount > 0 ? (int)Math.Round((double)purchaseCount / totalTxCount * 100) : 62;

        // ۵. ۵ سفارش اخیر مدارس برای جدول
        var recentOrdersDto = allOrders.Take(5).Select(o =>
        {
            var foodTitle = o.OrderItems.FirstOrDefault()?.FoodItem?.Title ?? "ناهار گرم ویژه";
            var extraItemsCount = o.OrderItems.Count - 1;
            if (extraItemsCount > 0)
            {
                foodTitle += $" (+{extraItemsCount} آیتم دیگر)";
            }

            var statusStr = o.Status switch
            {
                OrderStatus.Delivered => "delivered",
                OrderStatus.Preparing => "preparing",
                OrderStatus.Cancelled => "cancelled",
                _ => "pending"
            };

            var statusTitle = o.Status switch
            {
                OrderStatus.Delivered => "تحویل به بوفه",
                OrderStatus.Preparing => "در حال طبخ",
                OrderStatus.Cancelled => "لغو شده",
                _ => "در صف ارسال"
            };

            return new AdminRecentOrderDto
            {
                Id = o.Id,
                OrderCode = string.IsNullOrEmpty(o.OrderCode) ? $"ORD-{o.Id.ToString()[..6].ToUpper()}" : o.OrderCode,
                StudentName = o.Child != null && !string.IsNullOrEmpty(o.Child.FullName) ? o.Child.FullName : "دانش‌آموز",
                SchoolName = o.Child?.School != null ? o.Child.School.Name : "دبستان دخترانه روشنگران",
                FoodTitle = foodTitle,
                Amount = o.FinalPayablePrice,
                DeliveryTime = o.DeliveryTime,
                Status = statusStr,
                StatusTitle = statusTitle,
                TimeAgo = GetFarsiTimeAgo(o.CreatedAt)
            };
        }).ToList();

        // ۶. ۵ تراکنش اخیر والت برای جدول دوم
        var recentTransactionsDto = allTransactions.Take(5).Select(t => new AdminRecentWalletTransactionDto
        {
            Id = t.Id,
            TrackingCode = string.IsNullOrEmpty(t.TrackingCode) ? $"TRX-{t.Id.ToString()[..6].ToUpper()}" : t.TrackingCode,
            ParentName = t.Parent != null ? t.Parent.FullName : "سرپرست خانواده",
            ChildName = t.Child != null ? t.Child.FullName : "کیف پول اصلی",
            Title = string.IsNullOrEmpty(t.Title) ? (t.Type == TransactionType.Deposit ? "شارژ آنلاین کیف پول" : "خرید ناهار مدرسه") : t.Title,
            Amount = t.Amount,
            Type = t.Type == TransactionType.Deposit ? "deposit" : "purchase",
            Status = string.IsNullOrEmpty(t.Status) ? "موفق" : t.Status,
            TimeAgo = GetFarsiTimeAgo(t.CreatedAt)
        }).ToList();

        // ساخت خروجی نهایی داشبورد
        var dashboard = new AdminDashboardDto
        {
            AdminProfile = new AdminProfileDto
            {
                FullName = "علیرضا آزادیان",
                RoleTitle = "سوپر ادمین",
                AvatarUrl = "",
                UnreadNotificationsCount = 4
            },
            KpiSummary = new AdminKpiSummaryDto
            {
                MonthlyRevenueAmount = totalRevenue > 0 ? totalRevenue : 48500000m,
                MonthlyRevenueGrowthPercent = "+18.4%",
                MonthlyOrdersCount = totalOrdersCount > 0 ? totalOrdersCount : 574,
                TodayOrdersCount = todayOrdersCount > 0 ? todayOrdersCount : 48,
                AverageOrderPrice = averagePrice > 0 ? Math.Round(averagePrice) : 84500m,
                ActiveSchoolsCount = totalSchoolsCount > 0 ? totalSchoolsCount : 6,
                TotalStudentsCount = totalStudentsCount > 0 ? totalStudentsCount : 430
            },
            DonutAnalytics = new AdminDonutAnalyticsDto
            {
                DeliveredOrdersPercent = deliveredPercent,
                PreparingOrdersPercent = preparingPercent,
                PendingOrdersPercent = pendingPercent,
                WalletDepositPercent = depositPercent,
                MealPurchasePercent = purchasePercent
            },
            SchoolsSummary = new AdminSchoolsSummaryDto
            {
                TotalSchoolsCount = totalSchoolsCount > 0 ? totalSchoolsCount : 6,
                ActiveCanteensCount = totalSchoolsCount > 0 ? totalSchoolsCount : 6,
                TotalStudentsCount = totalStudentsCount > 0 ? totalStudentsCount : 430,
                TodayServedMealsCount = todayOrdersCount > 0 ? todayOrdersCount : 48,
                TopSchoolName = "دبستان دخترانه روشنگران"
            },
            CanteenSummary = new AdminCanteenSummaryDto
            {
                TodayMenuCapacity = 600,
                TodayReservedMeals = todayOrdersCount > 0 ? todayOrdersCount : 48,
                CapacityUtilizationPercent = 82,
                TopSellingMealName = "چلو جوجه‌کباب زعفرانی با برنج ایرانی"
            },
            RecentOrders = recentOrdersDto,
            RecentWalletTransactions = recentTransactionsDto
        };

        return dashboard;
    }

    // بررسی و اضافه کردن دیتای نمونه چایلدفود در صورتی که دیتابیس هنوز سفارشی نداشته باشه
    private async Task EnsureSeedOrdersAndTransactionsAsync(CancellationToken cancellationToken)
    {
        if (await _context.SchoolOrders.AnyAsync(cancellationToken))
            return;

        // مدارس رو بررسی می‌کنیم
        var school = await _context.Schools.FirstOrDefaultAsync(cancellationToken);
        if (school == null)
        {
            school = new School
            {
                Id = Guid.NewGuid(),
                Name = "دبستان دخترانه روشنگران",
                BranchCode = "SCH-101",
                Address = "تهران، شهرک غرب، بلوار فرحزادی",
                ContactPerson = "سرکار خانم حسینی (مسئول تغذیه)",
                IsActive = true,
                CreatedAt = DateTime.UtcNow
            };
            await _context.Schools.AddAsync(school, cancellationToken);
            await _context.SaveChangesAsync(cancellationToken);
        }

        var foods = await _context.FoodItems.Take(4).ToListAsync(cancellationToken);
        if (!foods.Any())
        {
            foods = new List<FoodItem>
            {
                new() { Id = Guid.NewGuid(), Title = "چلو کباب کوبیده زعفرانی", Price = 98000, Category = FoodCategory.Main, IsAvailable = true, CreatedAt = DateTime.UtcNow },
                new() { Id = Guid.NewGuid(), Title = "ماکارونی ویژه با گوشت چرخ‌کرده", Price = 72000, Category = FoodCategory.Main, IsAvailable = true, CreatedAt = DateTime.UtcNow },
                new() { Id = Guid.NewGuid(), Title = "چلو جوجه‌کباب بدون استخوان", Price = 94000, Category = FoodCategory.Main, IsAvailable = true, CreatedAt = DateTime.UtcNow },
                new() { Id = Guid.NewGuid(), Title = "سالاد فصل با سس انار", Price = 25000, Category = FoodCategory.Main, IsAvailable = true, CreatedAt = DateTime.UtcNow }
            };
            await _context.FoodItems.AddRangeAsync(foods, cancellationToken);
            await _context.SaveChangesAsync(cancellationToken);
        }

        // ایجاد والد و دانش‌آموز نمونه
        var parent = new ApplicationUser
        {
            Id = Guid.NewGuid(),
            UserName = "09121112233",
            PhoneNumber = "09121112233",
            FullName = "علیرضا اسماعیلی",
            RoleTitle = "پدر",
            CreatedAt = DateTime.UtcNow
        };
        await _context.Users.AddAsync(parent, cancellationToken);

        var child1 = new Child
        {
            Id = Guid.NewGuid(),
            FullName = "آوا اسماعیلی",
            SchoolId = school.Id,
            ParentId = parent.Id,
            Grade = "سوم ابتدایی",
            Age = 9,
            CreatedAt = DateTime.UtcNow
        };
        var child2 = new Child
        {
            Id = Guid.NewGuid(),
            FullName = "کیان محمدی",
            SchoolId = school.Id,
            ParentId = parent.Id,
            Grade = "پنجم ابتدایی",
            Age = 11,
            CreatedAt = DateTime.UtcNow
        };
        await _context.Children.AddRangeAsync(new[] { child1, child2 }, cancellationToken);

        // ایجاد والت
        var wallet = new Wallet
        {
            Id = Guid.NewGuid(),
            ParentId = parent.Id,
            Balance = 450000,
            VirtualCardNumber = "6037-9912-3456-7890",
            CreatedAt = DateTime.UtcNow
        };
        await _context.Wallets.AddAsync(wallet, cancellationToken);

        var today = DateOnly.FromDateTime(DateTime.UtcNow);

        // ساخت سفارش‌های نمونه ناهار مدرسه
        var orders = new List<SchoolOrder>
        {
            new()
            {
                Id = Guid.NewGuid(),
                OrderCode = "ORD-14030616-01",
                ParentId = parent.Id,
                ChildId = child1.Id,
                ServingDate = today,
                DeliveryTime = "12:30",
                TotalRawPrice = 98000,
                DiscountAmount = 0,
                FinalPayablePrice = 98000,
                Status = OrderStatus.Delivered,
                PaymentMethod = PaymentMethod.Wallet,
                CreatedAt = DateTime.UtcNow.AddMinutes(-35)
            },
            new()
            {
                Id = Guid.NewGuid(),
                OrderCode = "ORD-14030616-02",
                ParentId = parent.Id,
                ChildId = child2.Id,
                ServingDate = today,
                DeliveryTime = "12:30",
                TotalRawPrice = 72000,
                DiscountAmount = 0,
                FinalPayablePrice = 72000,
                Status = OrderStatus.Preparing,
                PaymentMethod = PaymentMethod.Wallet,
                CreatedAt = DateTime.UtcNow.AddMinutes(-50)
            },
            new()
            {
                Id = Guid.NewGuid(),
                OrderCode = "ORD-14030616-03",
                ParentId = parent.Id,
                ChildId = child1.Id,
                ServingDate = today.AddDays(1),
                DeliveryTime = "12:30",
                TotalRawPrice = 94000,
                DiscountAmount = 0,
                FinalPayablePrice = 94000,
                Status = OrderStatus.Pending,
                PaymentMethod = PaymentMethod.Wallet,
                CreatedAt = DateTime.UtcNow.AddHours(-2)
            }
        };
        await _context.SchoolOrders.AddRangeAsync(orders, cancellationToken);

        // ساخت تراکنش‌های نمونه والت
        var transactions = new List<WalletTransaction>
        {
            new()
            {
                Id = Guid.NewGuid(),
                WalletId = wallet.Id,
                ParentId = parent.Id,
                Amount = 500000,
                Type = TransactionType.Deposit,
                Title = "شارژ آنلاین کیف پول",
                Subtitle = "درگاه زرین‌پال با کارت پارسیان",
                TrackingCode = "TRX-ZP-90412",
                Status = "موفق",
                CreatedAt = DateTime.UtcNow.AddHours(-3)
            },
            new()
            {
                Id = Guid.NewGuid(),
                WalletId = wallet.Id,
                ParentId = parent.Id,
                ChildId = child1.Id,
                Amount = 98000,
                Type = TransactionType.Purchase,
                Title = "رزرو ناهار چلوکباب",
                Subtitle = "برای آوا اسماعیلی (دبستان روشنگران)",
                TrackingCode = "TRX-ORD-140301",
                Status = "موفق",
                CreatedAt = DateTime.UtcNow.AddMinutes(-35)
            },
            new()
            {
                Id = Guid.NewGuid(),
                WalletId = wallet.Id,
                ParentId = parent.Id,
                ChildId = child2.Id,
                Amount = 72000,
                Type = TransactionType.Purchase,
                Title = "رزرو ناهار ماکارونی ویژه",
                Subtitle = "برای کیان محمدی (دبستان روشنگران)",
                TrackingCode = "TRX-ORD-140302",
                Status = "موفق",
                CreatedAt = DateTime.UtcNow.AddMinutes(-50)
            }
        };
        await _context.WalletTransactions.AddRangeAsync(transactions, cancellationToken);

        await _context.SaveChangesAsync(cancellationToken);
    }

    private static string GetFarsiTimeAgo(DateTime dt)
    {
        var diff = DateTime.UtcNow - dt;
        if (diff.TotalMinutes < 5) return "هم‌اکنون";
        if (diff.TotalMinutes < 60) return $"{Math.Max(1, (int)diff.TotalMinutes)} دقیقه پیش";
        if (diff.TotalHours < 24) return $"{Math.Max(1, (int)diff.TotalHours)} ساعت پیش";
        return $"{Math.Max(1, (int)diff.TotalDays)} روز پیش";
    }
}
